packages/fxa-settings/src/pages/ResetPasswordRedesign/AccountRecoveryConfirmKeyOtp/index.tsx

```diff
@@ -0,0 +1,167 @@
+
+import AppLayout from '../../../components/AppLayout';
+import Banner, { BannerType } from '../../../components/Banner';
+import CardHeader from '../../../components/CardHeader';
+import InputText from '../../../components/InputText';
+import WarningMessage from '../../../components/WarningMessage';
+
+import {
+  AccountRecoveryConfirmKeyFormData,
+  AccountRecoveryConfirmKeyOtpProps,
+} from './interfaces';
+
+const AccountRecoveryConfirmKeyOtp = ({
+  accountResetToken,
+  code,
+  email,
+  emailToHashWith,
+  errorMessage,
+  estimatedSyncDeviceCount,
+  isSubmitting,
+  serviceName,
+  setErrorMessage,
+  setIsSubmitting,
+  verifyRecoveryKey,
+  token,
+  uid,
+}: AccountRecoveryConfirmKeyOtpProps) => {
+  const ftlMsgResolver = useFtlMsgResolver();
+  const location = useLocation();
+
+  const { getValues, handleSubmit, register, formState } =
+    useForm<AccountRecoveryConfirmKeyFormData>({
+      mode: 'onChange',
+      criteriaMode: 'all',
+      defaultValues: {
+        recoveryKey: '',
+      },
+    });
+
+  useEffect(() => {
+    GleanMetrics.resetPassword.recoveryKeyView();
+  }, []);
+
+  const onSubmit = () => {
+    setErrorMessage('');
+    // When users create their recovery key, the copyable output has spaces and we
+    // display it visually this way to users as well for easier reading. Strip that
+    // from here for less copy-and-paste friction for users.
+    const recoveryKey = getValues('recoveryKey').replace(/\s/g, '');
+
+    if (recoveryKey.length === 32 && isBase32Crockford(recoveryKey)) {
+      setIsSubmitting(true);
+      GleanMetrics.resetPassword.recoveryKeySubmit();
+      verifyRecoveryKey(recoveryKey);
+    } else {
+      // if the submitted key does not match the expected format,
+      // abort before submitting to the auth server
+      const localizedErrorMessage = getLocalizedErrorMessage(
+        ftlMsgResolver,
+        AuthUiErrors.INVALID_RECOVERY_KEY
+      );
+      setErrorMessage(localizedErrorMessage);
+      setIsSubmitting(false);
+    }
+  };
+
+  return (
+    <AppLayout>
+      <CardHeader
+        headingWithDefaultServiceFtlId="account-recovery-confirm-key-heading-w-default-service"
+        headingWithCustomServiceFtlId="account-recovery-confirm-key-heading-w-custom-service"
+        headingText="Reset password with account recovery key"
+        {...{ serviceName }}
+      />
+
+      {errorMessage && <Banner type={BannerType.error}>{errorMessage}</Banner>}
+
+      <FtlMsg id="account-recovery-confirm-key-instructions-2">
+        <p className="mt-4 text-sm">
+          Please enter the one time use account recovery key you stored in a
+          safe place to regain access to your Mozilla account.
+        </p>
+      </FtlMsg>
+      <WarningMessage
+        warningMessageFtlId="account-recovery-confirm-key-warning-message"
+        warningType="Note:"
+      >
+        If you reset your password and don't have account recovery key saved,
```

Comment:
Probably WIP but some of these don't have FTL IDs. This is also a straight quote instead of curly.

Reply:
The ftl id is passed in to the component here. The string already existed in the previous design, and I opted not to recreate new FTL entries since the text here is unchanged.